packages/backend/src/open_payments/payment/incoming/service.ts

```diff
@@ -135,7 +141,34 @@ async function createIncomingPayment(
     data: incomingPayment.toData(0n)
   })
 
-  return await addReceivedAmount(deps, incomingPayment, BigInt(0))
+  incomingPayment = await addReceivedAmount(deps, incomingPayment, BigInt(0))
+  if (!deps.config.pollIncomingPaymentCreatedWebhook) {
+    return incomingPayment
+  }
+
+  try {
+    const response = await poll({
+      request: async () => getApprovedOrCanceledIncomingPayment(deps, { id: incomingPayment.id }),
+      pollingFrequencyMs: deps.config.incomingPaymentCreatedPollFrequency,
+      timeoutMs: deps.config.incomingPaymentCreatedPollTimeout
+    })
+
+    if (response) return response
+    return IncomingPaymentError.ActionNotPerformed
+  }catch(err) {
+    return IncomingPaymentError.ActionNotPerformed
+  }
+
+}
+
+async function getApprovedOrCanceledIncomingPayment(deps: ServiceDependencies, options: GetOptions) {
+  const incomingPayment = await getIncomingPayment(deps, { id: options.id })
```

Comment:
I think what would be better instead of using `getIncomingPayment` is writing another query here. I want to avoid using `getIncomingPayment` because `addReceivedAmount` ends up doing the accounting lookup -> but we probably only want to do this once we know that the payment is approved or cancelled.

The query can include a check for where `approvedAt` or `cancelledAt` is not null explicitly (ie `IncomingPayment.query(...).whereNotNull(...)`).